PythonVisualizations/Stack.py

```diff
@@ -0,0 +1,322 @@
+            # create display objects for the associated Drawables
+            n.display_shape, n.display_val = self.createCellValue(
+                i, n.val, n.color)
+            n.color = self.canvas.itemconfigure(n.display_shape, 'fill')
+
+        self.window.update()
+
+    def makeButtons(self):
+        vcmd = (self.window.register(numericValidate),
+                '%d', '%i', '%P', '%s', '%S', '%v', '%V', '%W')
+
+        # numArguments decides the side where the button appears in the operations grid
+        pushButton = self.addOperation(
+            "Push", lambda: self.clickPush(), numArguments=1)  # validationCmd=vcmd)
+        popButton = self.addOperation(
+            "Pop", lambda: self.clickPop())
+        peekButton = self.addOperation(
+            "Peek", lambda: self.clickPeek())
+        newStackButton = self.addOperation(
+            "New", lambda: self.clickNewStack(), numArguments=1, validationCmd=vcmd)
+
+        return [pushButton, popButton, peekButton, newStackButton]
+
+    def validArgument(self):
+        entered_text = self.getArgument()
+        if entered_text:
+            if len(entered_text) <= self.maxArgWidth:
+                return entered_text
+            else:
+                self.setMessage("Error! {} value is too long".format(entered_text))
+
+    # Button functions
+    def clickPush(self):
+        val = self.validArgument()
+        if val is None:
+            # self.setMessage("Input value must be an integer from 0 to 99.")
+            self.setMessage("Input value must be of a string of length less than", self.maxArgWidth, "characters.")
+        else:
+            if len(self.list) >= self.size:
+                self.setMessage("Error! Stack is already full.")
+            else:
+                self.setButtonsStatus(DISABLED)
+                self.push(val)
+                self.setButtonsStatus(NORMAL)
+                self.setMessage("Value {} pushed!".format(val))
+        self.clearArgument()
+
+    def clickPop(self):
+        self.setButtonsStatus(DISABLED)
+        val = self.pop()
+        if val is None:
+            self.setMessage("Error! Stack is empty.")
+        else:
+            self.setMessage("Value {} popped!".format(val))
+        self.setButtonsStatus(NORMAL)
+
+    def clickPeek(self):
+        self.setButtonsStatus(DISABLED)
+
+        if len(self.list) <= 0:
+            self.setMessage("Error! Stack is empty.")
+        else:
+            val = self.list[-1].val
+            self.peek()
+            self.setMessage("Value {} is at the top of the stack!".format(val))
+        self.setButtonsStatus(NORMAL)
+
+    def clickNewStack(self):
+        val = self.validArgument()
+        val = int(val)
+        if val is None or 1 > val or val > self.MAX_SIZE:
+            self.setMessage("Error! Stack size must be an int between 1 and {}.".format(self.MAX_SIZE))
+
+        else:
+            stack.setButtonsStatus(DISABLED)
+            self.newStack(val)
```

Comment:
The logic on checking the user's input is flawed.  `validArgument()` will return the string the user typed or `None` if it is too long.  In the case of a request to create a new stack, we're expecting an integer string.  If `validArgument()` returns `None` then it passing it to `int()` will raise an exception.  The next check should be that the string contains only digits, otherwise `int()` will raise an exception. 
How about this:
```
    val = self.validArgument()
    if val is None:
        return
    if not val.isdigit():
        self.setMessage("New stack size must be a number")
        return
    val = int(val)
    if 1 > val or val > self.MAX_SIZE:
```
and so on.